src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -125,6 +126,24 @@ public Uri GraphDocumentationURL
             }
         }
 
+        private bool? enableLegacyPolyCurveBehavior;
+        /// <summary>
+        /// PolyCurve normal and direction behavior has been made predictable in Dynamo 3.0 and has therefore changed. 
+        /// This reflects whether legacy (pre-3.0) PolyCurve behavior is selected either in preference settings or in the workspace.
+        /// A workspace setting if exists, overrides the default preference setting. 
+        /// </summary>
+        public bool? EnableLegacyPolyCurveBehavior
```

Comment:
@aparajit-pratap I added comment but I can't find it now, so apologies if you already saw this:

what about adding a property to the Workspace to hold all of these configuration options that need to be present even in headless mode, IE if we don't have a view, or someone does not load the view for some reason, like you have done here - 
for example you could have an options object like:
`WorkspaceModel.LegacyBehaviorConfiguration` or you could consider moving the DynamoPreferencesData class to the Model.

Reply:
@mjkkirschner just so that I'm understanding you correctly, you're asking 2 things here:
1. Have an umbrella property that holds all configuration options 
2. Have this property at the Model level so it can be used in headless mode or at least have those that are relevant for headless mode

So I think I'm already doing (2)  I think this new property will be accessible headlessly as well. In fact what I was doing before was following the same thing as `ScaleFactor` setting, which for some reason is added to the `View` block in a DYN file, which doesn't make sense, which is why I moved this new option out of that.

With regards to (1), yes, we can definitely do that. It will be good to have a central place where we can add all such configuration-related things. My only point is, whether it be individual flags or individual options inside a `LegacyBehavior `type, that I can see them being in a state of flux - being added and removed over time as and when we decide to completely stop supporting specific legacy behaviors.

Reply:
About `DynamoPreferencesData`, I see these are the properties inside it and they are all inside the `View` block, which doesn't look like it's the right place for them; they seem that they are all (with the exception of maybe 2) applicable even outside the `View` block.
```
"View": {
    "Dynamo": {
      "ScaleFactor": 1.0,
      "HasRunWithoutCrash": true,
      "IsVisibleInDynamoLibrary": true,
      "Version": "3.0.0.5795",
      "RunType": "Automatic",
      "RunPeriod": "1000"
    },
```

Reply:
I think that at some point we started deserializing the view, even when the graph is executed in a model only context. I'm not sure when that change was finished, so maybe my concern about headless contexts is not completely valid, but I agree that the view doesn't seem like the correct place for these.

On the second concern - you could use a dictionary of options as the central location,  I think this would avoid the concern about adding and removing public members. Another idea is to make the members internal and serialize them by using the `[JsonProperty]` attribute.